assets/src/editor/events/ticket-assignments-manager/hooks/use-counts-manager.js

```diff
@@ -0,0 +1,141 @@
+		[ assignedState ]
+	);
+
+	const [ hasNoAssignments, noAssignmentsMessage, updatedAssignmentsCounts ] = useMemo( () => {
+		let howManyEmpty = 0;
+		const whichAreEmpty = { dates: 0, tickets: 0 };
+		const updatedCounts = cloneDeep( assignedCounts );
+		// first let's see if any of the current counts are 0 (but only not
+		// already toggled as empty)!
+		// instead of using whichAreEmpty as a ref, use assignedState (whether its empty or not) to decide whether we
+		// do this here or not. If assigned state is empty then we can do it.
+		if ( isEmptyAssignedState() ) {
+			const countCallback = ( accumulator, count ) => {
+				if ( count === 0 ) {
+					accumulator++;
+				}
+				return accumulator;
+			};
+			whichAreEmpty.dates = reduce( assignedCounts.dates, countCallback, 0 );
+			whichAreEmpty.tickets =	reduce( assignedCounts.tickets, countCallback, 0 );
+			howManyEmpty = whichAreEmpty.dates + whichAreEmpty.tickets;
+		}
+
+		if ( howManyEmpty > 0 ) {
+			return [ true, getMessage( whichAreEmpty ), updatedCounts ];
+		}
+
+		// still here?  Let's check whether the assignment being added or removed changes things.
+		const countCallback = ( assigning ) => ( emptyCount, ticketIds, dateId ) => {
+			if ( assigning ) {
+				ticketIds.forEach( ( ticketId ) => {
+					if ( ! assignedCounts.tickets[ ticketId ] ) {
+						whichAreEmpty.tickets--;
+						emptyCount--;
+					}
+					if ( typeof updatedCounts.tickets[ ticketId ] === 'undefined' ) {
+						updatedCounts.tickets[ ticketId ] = 0;
+					}
+					updatedCounts.tickets[ ticketId ]++;
+				} );
+				if ( ! assignedCounts.dates[ dateId ] ) {
+					whichAreEmpty.dates--;
+					emptyCount--;
+				}
+				if ( typeof updatedCounts.dates[ dateId ] === 'undefined' ) {
+					updatedCounts.dates[ dateId ] = 0;
+				}
+				updatedCounts.dates[ dateId ]++;
+			} else {
+				ticketIds.forEach( ( ticketId ) => {
+					if ( ! assignedCounts.tickets[ ticketId ] || assignedCounts.tickets[ ticketId ] - 1 === 0 ) {
+						whichAreEmpty.tickets++;
+						emptyCount++;
+					}
+					if ( typeof updatedCounts.tickets[ ticketId ] === 'undefined' ) {
+						updatedCounts.tickets[ ticketId ] = 0;
+					} else {
+						updatedCounts.tickets[ ticketId ]--;
+					}
+				} );
+				if ( ! assignedCounts.dates[ dateId ] || assignedCounts.dates[ dateId ] - 1 === 0 ) {
+					whichAreEmpty.dates++;
+					emptyCount++;
+				}
+				if ( typeof updatedCounts.dates[ dateId ] === 'undefined' ) {
+					updatedCounts.dates[ dateId ] = 0;
+				} else {
+					updatedCounts.dates[ dateId ]--;
+				}
+			}
+			return emptyCount;
+		};
+		howManyEmpty = reduce( assignedState.assigned, countCallback( true ), howManyEmpty );
+		howManyEmpty = reduce( assignedState.removed, countCallback( false ), howManyEmpty );
+
+		// ensure whichAreEmpty don't go lower than zero
+		whichAreEmpty.dates = Math.max( 0, whichAreEmpty.dates );
+		whichAreEmpty.tickets = Math.max( 0, whichAreEmpty.tickets );
```

Comment:
It is possible for `whichAreEmpty` values to end up negative depending on how many tickets/dates are being processed and what are assigned or not assigned.  Because `whichAreEmpty` is used in determining what message is printed, it's easier to just make sure that we convert negative to zero here.